dopamine/replay_memory/circular_replay_buffer.py

```diff
@@ -297,38 +283,23 @@ def _add(self, observation, action, reward, terminal, *args):
       terminal: A uint8 acting as a boolean indicating whether the transition
                 was terminal (1) or not (0).
       *args: All the elements in a transition.
-
-    Raises:
-      ValueError: If `transition_index` is not in the range
-        [0, len(self._trajectories)].
     """
-    trajectory_index = self._get_current_trajectory()
-    if not 0 <= trajectory_index < len(self._trajectories):
-      raise ValueError(
-          '`trajectory_index` must be in the '
-          'range [0, {}[. Given {} instead.'.format(
-              len(self._trajectories), trajectory_index))
-    trajectory = self._trajectories[trajectory_index]
-    trajectory.append(
+    self._trajectory.append(
```

Comment:
:-)  This was a lot of work you did and undid to separate out "adding trajectories" and "using trajectories" into separate CLs.  It would have been ok to submit them both as part of the same patch.  The last one would have been a little easier to review had they been pushed together because its intended use would have been more clear.

Reply:
Yes that's because I found a way to simplify the logic using the local variables library we created  later one, only after I sent the 2 pull request.
Agreed with you on this point.